src/PrMerger.js

```diff
@@ -54,20 +53,30 @@ class PrMerger {
     // there is a PR still-in-merge.
     async runStep() {
         Logger.info("runStep running");
-        const finalizer = await this._current();
-        const prList = await this._getPRList(finalizer);
+        // all repository tags
+        this._tags = await GH.getTags();
+        this._prList = await GH.getPRList();
+        this._logPRList("PRs got from GitHub: ");
+
+        await this._cleanTags();
+
+        const stagingPr = await this._current();
+        await this._preparePRList(stagingPr);
 
         this.total = 0;
-        while (prList.length) {
+        let suspendedEarlier = false;
+        while (this._prList.length) {
             try {
-                const pr = prList.shift();
+                const rawPr = this._prList.shift();
                 this.total++;
-                const result = await pr.anubisProcessor.process();
-                if (!this.prDone(pr.anubisProcessor, result))
-                    return true;
+                let pr = new PullRequest(rawPr, suspendedEarlier);
+                const result = await pr.process();
+                suspendedEarlier = suspendedEarlier || result.suspended();
```

Comment:
Not quite done: If _doProcess() throws, the process() method now _throws_  instead of returning a valid result. I do not think we want process() to throw. At the very least, its previous version caught exceptions instead of throwing IIRC. Note that the "finally" block does not swallow exceptions like the "catch" block does. You probably want both blocks, with the "catch" block returning some reasonable result after logging the exception.

Reply:
> If _doProcess() throws, the process() method now throws instead of returning a valid result.

That is what we need: this exception should go into the caller PrMerger.runStep() in order to go ahead with another PR.

>  At the very least, its previous version caught exceptions instead of throwing

No, its previous version (without 'finally') caught the exception, flushed pending labels and rethrew the exception.

Reply:
Got it. Sorry for the noise.